common/reactive/src/main/java/io/helidon/common/reactive/OriginThreadPublisher.java

```diff
@@ -24,107 +24,115 @@
-
-    private AtomicLong nextCount = new AtomicLong();
+    private final BlockingQueue<T> queue = new ArrayBlockingQueue<>(256);
+    private final AtomicLong nextCount = new AtomicLong();
     private volatile long reqCount = 0;
 
     /**
      * Create same thread publisher.
      *
-     * @param semaphore      the semaphore to indicate the amount of requested data. The owner of this publisher
-     *                       is responsible to send the data as determined by the semaphore (i.e., to properly
-     *                       acquire a permission to send the data; to not send when the number of permits is zero).
-     * @param referenceQueue the reference queue to associate the
-     *                       {@link io.helidon.webserver.ReferenceHoldingQueue.ReleasableReference} instances with
+     * @param semaphore the semaphore to indicate the amount of requested data.
+     * The owner of this publisher is responsible to send the data as determined
+     * by the semaphore (i.e., to properly acquire a permission to send the
+     * data; to not send when the number of permits is zero).
      */
-    OriginThreadPublisher(UnboundedSemaphore semaphore, ReferenceHoldingQueue<ByteBufRequestChunk> referenceQueue) {
+    public OriginThreadPublisher(UnboundedSemaphore semaphore) {
         this.semaphore = semaphore;
-        this.referenceQueue = referenceQueue;
     }
 
     /**
      * Create same thread publisher.
-     *
-     * @param referenceQueue the reference queue to associate the
-     *                       {@link io.helidon.webserver.ReferenceHoldingQueue.ReleasableReference} instances with
      */
-    OriginThreadPublisher(ReferenceHoldingQueue<ByteBufRequestChunk> referenceQueue) {
-        this(new UnboundedSemaphore(), referenceQueue);
+    public OriginThreadPublisher() {
+        this(new UnboundedSemaphore());
     }
 
     @Override
-    public void subscribe(Flow.Subscriber<? super DataChunk> originalSubscriber) {
+    public void subscribe(Subscriber<? super T> originalSubscriber) {
         if (!hasSingleSubscriber.compareAndSet(false, true)) {
-            originalSubscriber.onError(new IllegalStateException("Only single subscriber is allowed!"));
+            originalSubscriber.onError(new IllegalStateException(
+                    "Only single subscriber is allowed!"));
             return;
         }
 
         singleSubscriber = originalSubscriber;
 
         reentrantLock.lock();
         try {
-            originalSubscriber.onSubscribe(new Flow.Subscription() {
+            originalSubscriber.onSubscribe(new Subscription() {
 
                 private boolean nexting;
 
                 @Override
                 public void request(long n) {
                     if (n <= 0) {
-                        error(new IllegalArgumentException("[3.9] Illegal value requested: " + n));
+                        error(new IllegalArgumentException(
```

Comment:
Please do not line break

Reply:
I fixed a lot of the over re-formats, including this one.